ttnn/ttnn/multi_device.py

```diff
@@ -443,7 +446,30 @@ def __init__(self, mesh_device: MeshDevice):
         self.mesh_device = mesh_device
 
     def compose(self, tensor: ttnn.Tensor) -> List["torch.Tensor"]:
-        return [ttnn.to_torch(tt_input_tensor) for tt_input_tensor in ttnn.get_device_tensors(tensor)]
+        return [
+            ttnn.to_torch(tt_input_tensor, mesh_composer=None) for tt_input_tensor in ttnn.get_device_tensors(tensor)
+        ]
+
+
+@contextlib.contextmanager
+def distribute(mesh_mapper: TensorToMesh = None, mesh_composer: MeshToTensor = None):
```

Comment:
Should this be a union instead? You always want one. You can't accept 0 or 2.

Reply:
if functools.partial accepts union too, it makes this code even cleaner

Reply:
thanks for the suggestion. I've updated.